src/parse.cpp

```diff
@@ -224,47 +224,44 @@ void my_plugin::write_thread_category(const falcosecurity::table_entry& thread_e
 bool my_plugin::parse_new_process_event(
         const falcosecurity::parse_event_input& in) {
     // get tid
-    int64_t thread_id = in.get_event_reader().get_tid();
-    if(thread_id <= 0)
-    {
-        SPDLOG_INFO("unknown thread id for event num '{}' with type '{}'",
-                    in.get_event_reader().get_num(),
-                    int32_t(in.get_event_reader().get_type()));
-        return false;
-    }
+    auto thread_id = in.get_event_reader().get_tid();
 
     // compute container_id from tid->cgroups
     auto& tr = in.get_table_reader();
 
     // retrieve the thread entry associated with this thread id
-    auto thread_entry = m_threads_table.get_entry(tr, thread_id);
-
-    std::shared_ptr<container_info> info = nullptr;
-    auto container_id = compute_container_id_for_thread(thread_entry, tr, info);
-
-    // store container_id
-    auto& tw = in.get_table_writer();
-    m_container_id_field.write_value(tw, thread_entry,
-                                     (const char*)container_id.c_str());
-
-    if (info != nullptr) {
-        // Since the matcher also returned a container_info,
-        // it means we do not expect to receive any metadata from the go-worker,
-        // since the engine has no listener SDK.
-        // Just send the event now.
-        nlohmann::json j(info);
-        generate_async_event(j.dump().c_str(), true, ASYNC_HANDLER_DEFAULT);
-
-        // Immediately cache the container metadata
-        m_containers[info->m_id] = info;
-    }
-
-    // Write thread category field
-    if (container_id != HOST_CONTAINER_ID) {
-        write_thread_category(thread_entry, tr, tw);
+    try {
+    	auto thread_entry = m_threads_table.get_entry(tr, thread_id);
+
+    	std::shared_ptr<container_info> info = nullptr;
+    	auto container_id = compute_container_id_for_thread(thread_entry, tr, info);
+
+    	// store container_id
+    	auto& tw = in.get_table_writer();
+    	m_container_id_field.write_value(tw, thread_entry, container_id);
+
+    	if (info != nullptr) {
+        	// Since the matcher also returned a container_info,
+        	// it means we do not expect to receive any metadata from the go-worker,
+        	// since the engine has no listener SDK.
+        	// Just send the event now.
+        	nlohmann::json j(info);
+        	generate_async_event(j.dump().c_str(), true, ASYNC_HANDLER_DEFAULT);
+
+        	// Immediately cache the container metadata
+        	m_containers[info->m_id] = info;
+    	}
+
+    	// Write thread category field
+    	if (container_id != HOST_CONTAINER_ID) {
+        	write_thread_category(thread_entry, tr, tw);
+    	}
+    	return true;
+    } catch (falcosecurity::plugin_exception &e) {
+      	SPDLOG_ERROR("cannot attach container_id to new process event for the thread id '{}': {}",
```

Comment:
This is actually an error since we are parsing a `process` event (ie: execve, clone and so on), therefore `thread_id` MUST be present.